### PR TITLE
Model validation for FS models

### PR DESCRIPTION
**Pull request by aliazzzdat**
FS models cannot predict as pyfunc models yet (issue #70), thus we can't use MLflow Evaluate with FS models.

However, MLflow Evaluate can take a function instead of a pyfunc model_uri. 
This means we can call an FS model in the function and use this function in the Evaluate method.

Although this is a good solution, we still cannot enable baseline comparison, as we need to provide a model_uri of a pyfunc model (so this means it's not possible to compare a new FS model with a baseline FS model).

### CODE REVIEW

**Comment by arpitjasa-db** [comment]
@aliazzzdat thanks for all the changes! Looks like some tests are failing, can we fix those and also do a full instantiation of Stacks to confirm everything is still working as expected? Version changes can be tricky, and want to make sure we don't break anything

File: tests/example-project-configs/aws/aws-github.json [hunk 1]
```diff
@@ -4,12 +4,12 @@
   "input_project_name": "example-aws-github-project",
   "input_cloud": "aws",
   "input_cicd_platform": "github_actions",
-  "input_databricks_staging_workspace_host": "https://your-staging-workspace.cloud.databricks.com",
-  "input_databricks_prod_workspace_host": "https://your-prod-workspace.cloud.databricks.com",
+  "input_databricks_staging_workspace_host": "https://e2-demo-field-eng.cloud.databricks.com/",
```
Comment (arpitjasa-db) [inline tests/example-project-configs/aws/aws-github.json:7]: Revert changes to this file please

**Comment by aliazzzdat** [comment]
The batch inference works as expected once we create the feature_store_inference_input table 
Screenshot 2024-07-19 at 21 54 34

File: .gitignore [hunk 2]
```diff
@@ -11,4 +11,4 @@
 __pycache__/
 .cache
 *.pyc
-mlops-stacks.iml
+mlops-stacks.iml
```
Comment (arpitjasa-db) [inline .gitignore:14]: nit newline

File: tests/example-project-configs/aws/aws-github-fs.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (arpitjasa-db) [inline tests/example-project-configs/aws/aws-github-fs.json:1]: What's the purpose of this file? We don't run tests using this right?

Comment (aliazzzdat) [reply tests/example-project-configs/aws/aws-github-fs.json:1]: It's just the file I used to generate the template to test whether the changes were working. It's the same as the other one, except it generates an FS model.

File: .gitignore (hunk 2 above)
Comment (aliazzzdat) [reply .gitignore:14]: I added and removed  a file in my local .gitignore, the changes are just the space removed after .iml
Should we add *.iml in the file or do you want me revert the changes (adding the space back) ?

Comment (arpitjasa-db) [reply .gitignore:14]: Let's revert the changes

File: tests/example-project-configs/aws/aws-github-fs.json (hunk 3 above)
Comment (arpitjasa-db) [reply tests/example-project-configs/aws/aws-github-fs.json:1]: Hmm we don't really get much utility out of it except specify FS as yes, which if we did for each option, end up with tens of example configs. I think we should just remove this file then

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/validation/notebooks/ModelValidation.py.tmpl [hunk 4]
```diff
@@ -255,8 +257,83 @@ def log_to_model_description(run, success):
...
+
+data = rounded_taxi_data(data)
+
+
+{{ end }}
+
 # COMMAND ----------
 
+{{ if (eq .input_include_feature_store `yes`) }}
+###########################################################################################
```
Comment (arpitjasa-db) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/validation/notebooks/ModelValidation.py.tmpl:315]: Why do we have these lines of `#`?

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/validation/notebooks/ModelValidation.py.tmpl [hunk 5]
```diff
@@ -277,7 +354,11 @@ with mlflow.start_run(
 
     try:
         eval_result = mlflow.evaluate(
+            {{ if (eq .input_include_feature_store `yes`) }}
+            get_fs_model,
```
Comment (arpitjasa-db) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/validation/notebooks/ModelValidation.py.tmpl:358]:
```suggestion
            model=get_fs_model,
```

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/validation/notebooks/ModelValidation.py.tmpl (hunk 4 above)
Comment (aliazzzdat) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/validation/notebooks/ModelValidation.py.tmpl:315]: yes it's leftover comments that I put in place when I showed the fix to the FS team, I'm going to remove them

**Review by arpitjasa-db — APPROVED** [review]
Thanks for all the changes @aliazzzdat !

**Pull request merged**